netns/netnsinterface.go

```diff
@@ -0,0 +1,9 @@
+package netns
+
+type NetnsInterface interface {
```

Comment:
This is a Java OOP pattern 👎🏼. In Go, _we_ do not define the interfaces when we define the types (because we don't need to!). The _consumer_ of our types is free to define the interface representing the (subset of the) behavior of our types that they need to use abstractly, and our types will implicitly implement their interface.